linera-chain/src/manager.rs

```diff
@@ -121,12 +121,18 @@ pub struct ChainManager {
     /// validator).
     #[debug(skip_if = Option::is_none)]
     pub locked: Option<ValidatedBlockCertificate>,
+    /// These are the published blobs belonging to the locked block.
+    #[debug(skip_if = BTreeMap::is_empty)]
+    pub locked_published_blobs: BTreeMap<BlobId, Blob>,
```

Comment:
Isn't this part of the `locked` certificate? i.e. there's a block inside and `published_blobs` where part of it. Instead of putting it as a separate field, with a separate lifecycle, I'd extract the blobs from the `locked` certificate when necessary. Otherwise we need to add logic that manages the lifecycle of this field seperately from the `locked` cert.

Reply:
No, the certificate itself doesn't contain the blobs, only the hashes.

Reply:
Maybe it should contain them though? They're _published_, so it's the first time validators would see them, right?

Reply:
But the whole point of blobs is to move them _out_ of the blocks themselves.

Reply:
Isn't it the case that when a validator receives a block proposal, containing published (I'm assuming this means _new_) blobs, he will have to request them anyway?

Reply:
Yes, that's why we send them along in the `BlockProposal`, which contains the `Block` and the blobs. But the block itself doesn't contain the blobs, and the certificate contains only the block.

I guess this is just a matter of how our data types are organized. We _could_ have a `CertificateIncludingPublishedBlobs` type.

An argument against that would be that we might want to allow publishing lots of large blobs in a single block, that would exceed the gRPC message limits. The idea is that _blocks_ (and thus _certificates_) are small (always fit in the message limit), but blobs don't have to be.
So in the future we should rather go in the opposite direction and treat them separately everywhere, i.e. _not_ send them along in the `BlockProposal`, and not have them stored in a single `RegisterView<ChainManager>`.

Reply:
OK, clearly this doesn't belong to this PR. But. I think we can rethink how we structure certificates and different blocks (validated, confirmed) and I'll be doing that soon. Currently, due to a tech debt, all different variants contain the same data while they don't need to or maybe even shouldn't.